src/hooks/useCryptoBalances.ts

```diff
@@ -0,0 +1,184 @@
+          tokenAddress: CRYPTO_CONFIG[1].tokenAddress || '',
+          chainId: CRYPTO_CONFIG[1].celoChainId || 0,
+          balance: usdcCeloBalance.balance
+            ? `${usdcCeloBalance.balance.toLocaleString()} USDC`
+            : '0 USDC',
+          usdValue: usdcCeloBalance.balance
+            ? `$${usdcCeloBalance.balance.toLocaleString()}`
+            : '$0',
+          fullBalance: usdcCeloBalance.balance
+            ? usdcCeloBalance.balance.toString()
+            : '0',
+          currencyIcon: CRYPTO_CONFIG[1].icon,
+          networkName: 'Celo',
+          isLoading: usdcCeloBalance.isLoading,
+        },
+        {
+          ...CRYPTO_CONFIG[2], // USDT Celo
+          tokenAddress: CRYPTO_CONFIG[2].tokenAddress || '',
+          chainId: CRYPTO_CONFIG[2].celoChainId || 0,
+          balance: usdtCeloBalance.balance
+            ? `${usdtCeloBalance.balance.toLocaleString()} USDT`
+            : '0 USDT',
+          usdValue: usdtCeloBalance.balance
+            ? `$${usdtCeloBalance.balance.toLocaleString()}`
+            : '$0',
+          fullBalance: usdtCeloBalance.balance
+            ? usdtCeloBalance.balance.toString()
+            : '0',
+          currencyIcon: CRYPTO_CONFIG[2].icon,
+          networkName: 'Celo',
+          isLoading: usdtCeloBalance.isLoading,
+        },
+      ]
+    } else if (selectedNetwork === 'Arbitrum') {
+      return [
+        {
+          ...CRYPTO_CONFIG[1], // USDC Arbitrum
+          tokenAddress: CRYPTO_CONFIG[1].arbitrumTokenAddress || '',
+          chainId: CRYPTO_CONFIG[1].arbitrumChainId || 0,
+          balance: usdcArbitrumBalance.balance
+            ? `${usdcArbitrumBalance.balance.toLocaleString()} USDC`
+            : '0 USDC',
+          usdValue: usdcArbitrumBalance.balance
+            ? `$${usdcArbitrumBalance.balance.toLocaleString()}`
+            : '$0',
+          fullBalance: usdcArbitrumBalance.balance
+            ? usdcArbitrumBalance.balance.toString()
+            : '0',
+          currencyIcon: CRYPTO_CONFIG[1].icon,
+          networkName: 'Arbitrum',
+          isLoading: usdcArbitrumBalance.isLoading,
+        },
+        {
+          ...CRYPTO_CONFIG[2], // USDT Arbitrum
+          tokenAddress: CRYPTO_CONFIG[2].arbitrumTokenAddress || '',
+          chainId: CRYPTO_CONFIG[2].arbitrumChainId || 0,
+          balance: usdtArbitrumBalance.balance
+            ? `${usdtArbitrumBalance.balance.toLocaleString()} USDT`
+            : '0 USDT',
+          usdValue: usdtArbitrumBalance.balance
+            ? `$${usdtArbitrumBalance.balance.toLocaleString()}`
+            : '$0',
+          fullBalance: usdtArbitrumBalance.balance
+            ? usdtArbitrumBalance.balance.toString()
+            : '0',
+          currencyIcon: CRYPTO_CONFIG[2].icon,
+          networkName: 'Arbitrum',
+          isLoading: usdtArbitrumBalance.isLoading,
+        },
+      ]
+    } else if (selectedNetwork === 'Arbitrum Sepolia') {
```

Comment:
Manual filtering like this isn't an efficient way to address this, when we have 10 chains we then have 10 different branches of IFs ?